python/dgl/graphbolt/itemset.py

```diff
@@ -16,25 +16,31 @@ def is_scalar(x):
 
 
 class ItemSet:
-    r"""A wrapper of iterable data or tuple of iterable data.
+    r"""A wrapper of sequential data or tuple of sequential data.
+
+    This class requires each input item to be sequential (and, of course,
+    iterable), meaning that each item must have implemented `__getitem__` which
+    supports fetching a data for a given index, and `__len__` which is expected
+    to return the size of the item.
 
-    All itemsets that represent an iterable of items should subclass it. Such
-    form of itemset is particularly useful when items come from a stream. This
-    class requires each input itemset to be iterable.
 
     Parameters
     ----------
-    items: Union[int, Iterable, Tuple[Iterable]]
-        The items to be iterated over. If it is a single integer, a `range()`
-        object will be created and iterated over. If it's multi-dimensional
-        iterable such as `torch.Tensor`, it will be iterated over the first
-        dimension. If it is a tuple, each item in the tuple is an iterable of
-        items.
+    items: Union[int, torch.Tensor, Sequence, Tuple[Sequence]]
```

Comment:
`Tensor` is always sequence-able? If yes, is it better to remove `torch.Tensor` from the list?

Reply:
For simplicity, could we replace `Sequence` with `Tensor` directly? Namely, we just support integer and tensor only. Do we have any requirement for `Sequence` besides `Tensor`?

Reply:
`numpy.Array`?

Reply:
any specific testcase for `Sequence` besides `Tensor`?

Reply:
In general, I'd like to make `ItemSet` be as simple as possible: limited but clear, sufficient support for the use case we're facing for now or in near future. This is one of the motivation of we initiated the refactor.

Reply:
Agreed.

Reply:
But in this case, should we renam the parameter `item` to `tensor`? Furthermore, should we rename this class to `TensorWrapper` or something like that? Why would we introduce the term `item` if the item must be a tensor?

Reply:
`Item` mean what the data is, concept-wise noun, and `Tensor` is the detailed or implemented data type of the `Item`. So I think `ItemSet`is ok. no need to change it to TensorXXX